### PR TITLE
Add Bitnami stacks and blank project for use with the generators

### PR DESCRIPTION
**Pull request by amrecio**
### What does this PR do?

Add several Bitnami Stacks (Express 4.14.0, Swift 3.0-PREVIEW-6, Laravel 5.2.31, Rails 5.0.0.1, CodeIgniter 3.1.0, Symfony 3.1.3, and Play 2.55), and a "blank" sample application intended to be populated by the generators, which can be run as a Che command.
### What issues does this PR fix or reference?

It doesn't reference an issue.
### New behavior

Express stack added to Stack Library. Express sample added to samples.
### PR type
- [X] Minor change = no change to existing features or docs
- [ ] Major change = changes existing features or docs
### Minor change checklist
- [ ] New API required?
- [ ] API updated
- [ ] Tests provided / updated
- [ ] Tests passed
### Major change checklist
- [ ] New API required?
- [ ] API updated
- [ ] Documentation provided (include here or link to docs)
- [ ] Tests provided / updated
- [ ] Tests passed

Signed-off-by: Alvaro Manuel Recio Perez amrecio@gmail.com

### CODE REVIEW

**Comment by slemeur** [comment]
Thanks for your PR @amrecio ! We will review the stack asap. 
@ashumilova + @eivantsov : could you please give a look?

File: ide/che-core-ide-stacks/src/main/resources/images/type-bitnami.svg [hunk 1]
```diff
@@ -0,0 +1,177 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg
```
Comment (slemeur) [inline ide/che-core-ide-stacks/src/main/resources/images/type-bitnami.svg:11]: This icon is missing a licence header. 

You can give a look at this one:
https://github.com/eclipse/che/blob/master/ide/che-core-ide-stacks/src/main/resources/images/type-che.svg?short_path=2182d77#L2-L13

**Comment by ashumilova** [comment]
The stack and sample look good.
About icon - in Inkskape it can be saved as plain or optimized SVG to avoid unnecessary inkscape tags and metadata.

**Comment by amrecio** [comment]
Thanks for your comments! I'll update the icon to fix the issues.

**Comment by amrecio** [comment]
@ashumilova : I've updated the icon and saved it as "plain SVG". I left the metadata that it left but let me know if I should remove it too.

**Comment by ashumilova** [comment]
@amrecio  It's fine, thank you

**Comment by TylerJewell** [comment]
@slemeur - as part of this merge, once it is done, can you have James add the appropriate Tutorial pages to the Eclipse Che docs sections for each of these types?  Let's have a good summary on the docs page and then a link over to the bitnami pages where the full tutorial is provided.

**Comment by slemeur** [comment]
yes. and we will be more exhaustive on the components and tags as well.

**Comment by riuvshin** [comment]
this issue should be back ported to https://github.com/eclipse/che/tree/4.7.x branch for release 4.7.1
Please note that in current master model is different from one we have in 4.7.x so It may require to different PRs

**Comment by ashumilova** [comment]
Here is the adaption of current PR to model changes https://github.com/eclipse/che/pull/2345

**Comment by amrecio** [comment]
I've updated the PR to add several Bitnami stacks. We're having some issues with Docker Hub and a few images are still queued for building but everything should work once they're ready.

**Comment by slemeur** [comment]
@amrecio Could you update the PR with the latest changes?

**Comment by amrecio** [comment]
Yes, sorry. I merged the PR into master instead of the right branch. I'll update it right now.

**Comment by amrecio** [comment]
I've updated the PR and squashed the commits with the latest changes.

**Comment by slemeur** [comment]
Thanks a lot !

**Comment by amrecio** [comment]
@ashumilova Done! I kept the Bitrock copyright header in the SVG icon but let me know if I should use Codenvy's instead.

**Comment by riuvshin** [comment]
changed milestone to 5.0.0-M1 as we already ship 4.7.2 with those changes.

**Comment by ashumilova** [comment]
@amrecio it's ok, lets keep Bitrock copyright, but for Che to build and pass license check successfully need to add following to che-core-ide-stacks/pom.xml:
`<build>`
`<plugins>`
            `<plugin>`
                `<groupId>com.mycila</groupId>`
                `<artifactId>license-maven-plugin</artifactId>`
                `<configuration>`
                    `<excludes>`
                        `<exclude>**/type-bitnami.svg</exclude>`
                   `</excludes>`
               `</configuration>`
            `</plugin>`
       `</plugins>`
  `</build>`

**Comment by amrecio** [comment]
@ashumilova After fixing a wrong rebase on my part, the PR is ready, including your patch to `che-core-ide-stacks/pom.xml`. Please let me know if any other change is needed.

**Comment by ashumilova** [comment]
Thank you! Merged.

**Pull request merged**